src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -282,10 +291,16 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
     if task.trigger:
         source_triggers = task.trigger.source_triggers
         base_image_trigger = task.trigger.base_image_trigger
-        if commit_trigger_enabled or source_triggers is not None:
-            status = None
+        if (commit_trigger_enabled or pull_request_trigger_enabled or source_triggers) is not None:
+            #using a set to prevent duplicate event types per trigger
+            _events = set()
+            if source_triggers:
+                _events = set(source_triggers[0].source_trigger_events)
             if commit_trigger_enabled is not None:
-                status = TriggerStatus.enabled.value if commit_trigger_enabled else TriggerStatus.disabled.value
+                _events.add(SourceTriggerEvent.commit.value)
+            if pull_request_trigger_enabled is not None:
```

Comment:
Is this a valid check? What is the value is `False` .. Do we still add to events ?

Reply:
We updated the design to allow the user to enable commit and pull request events separately. If the value is false, then we do not add that event type to the trigger's event list.